react/javascript/demo-external-api/src/components/OauthProvider.jsx

```diff
@@ -0,0 +1,117 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+import React, { createContext, useContext, useState } from 'react'
+import PropTypes from 'prop-types'
+import { useHistory, useLocation } from 'react-router-dom'
+import { ExtensionContext2 } from '@looker/extension-sdk-react'
+
+export const OauthContext = createContext({})
+
+const authUrl = 'https://accounts.google.com/o/oauth2/v2/auth'
+
+/**
+ * Default push state handler.
+ * props object has one or two properties
+ * state - always present and represents current state
+ * token - only present for login and logout
+ *       - login - valid token
+ *       - logout - always undefined
+ * Push state handler gets called on three occassions
+ * 1. to get the current token - no token so get from state
+ * 2. to login - token property present so add token to state
+ * 3. to logout - token property present but undefined - remove from state.
+ */
+const defaultPushStateHandler = (props) => {
+  const { token, state } = props
+  return Object.prototype.hasOwnProperty.call(props, 'token') ? token : state
+}
+
+/**
+ * Oauth provider that stores an OAUTH2 access token in history push state.
+ * This example uses Googles implicit OAUTH2 flow.
+ */
+export const OauthProvider = ({
+  children,
+  clientId,
+  scopes,
+  pushStateHandler = defaultPushStateHandler,
+  loggedInRoute = '/home',
+  loggedOutRoute = '/',
+}) => {
+  const { extensionSDK } = useContext(ExtensionContext2)
+  const [loggingIn, setLoggingIn] = useState(false)
+  const history = useHistory()
+  const { state } = useLocation()
+  const token = pushStateHandler({ state })
+
+  /**
+   * OAUTH2 authentication.
+   */
+  const signIn = async () => {
+    try {
+      setLoggingIn(true)
+      const response = await extensionSDK.oauth2Authenticate(authUrl, {
+        client_id: clientId,
+        scope: scopes,
+        response_type: 'token',
+      })
+      // eslint-disable-next-line camelcase
```

Comment:
We should probably enable `ignoreDestructuring: true` in projects that use the API to avoid having to one-off these sorts of things.

Reply:
I've had to use
`camelcase: ['error', { ignoreDestructuring: true, properties: 'never' }],`